docs/rfc/0003-collaborative-objects.md

```diff
@@ -0,0 +1,649 @@
+This construction, along with sorting by actor IDs in the case of a tie, allows
+us to place operations in a total order which respects causality. i.e if I add
+an operation then no operation that I could have observed at the time I made
+the operation will come after it in the log.
+
+Automerge defines a number of operations along with merge semantics for those
+operations. More detail on that can be found in [the implementation](https://github.com/automerge/automerge)
+and in [the paper](https://arxiv.org/abs/1608.03960).
+
+Operations in automerge are transported in batches called "changes". Each 
+change references zero or more changes it depends on via their hash. In this 
+manner automerge is similar to git in that it's a hash linked graph of changes.
+
+Despite all the complexity under the hood, the API of automerge is relatively
+simple. Automerge works in terms of "documents", a document is a single log of
+changes. Every time you modify an automerge document you generate a new entry 
+for the change log. Each change is just some bytes. When you receive changes 
+from other actors you just pass these changes (which, again, are just bytes) to
+automerge to add to the change log. The end result is that you load a bunch of
+binary changes and get back a JSON object.
+
+There are some subtleties around preserving user intent when modifying
+documents, but these are not too onerous.
+
+### Object Trees
+
+Given that automerge changes are a hash linked graph, we can map them to Git.
+We do so by wrapping each change in a commit. The commit points at a tree with
+the following layout
+
+```
+.
+|--manifest
+|--schema
+|  |--schema.json
+|  |--migrations
+|  |  |--<migration hash>.json
+|  |  |--<migration hash>.json
+|--change
+|--author
+```
+
+This tree contains a single change to a collaborative object. We will go into
+more details shortly. Any direct dependencies of this change are encoded in the 
+same manner and become the parents of this commit. This allows us to
+reconstruct the automerge depdency graph. 
+
+Along with the dependencies of the commit we also need to add the commit of the
+identity which created this commit. We need this identity to validate
+signatures and by making the commit a parent we ensure that git will replicate
+it for us.
+
+
+#### `manifest`
+
+The manifest is a TOML file containing some metadata about the object.
+Specifically it will contain:
+
+- `id`, a UUID, generated at the time the object is created
+- `typename`, discussed above
+- `history_type`, always `"automerge"`, this is here to allow for different
+  CRDT implementations in future.
+
+
+Each object is also created with a JSON schema. The schema is represented by an
+initial `schema.json` and a series of schema migrations which extend that
+initial schema. Schema migrations will not be addressed in detail in this RFC
+but we will show their feasibility.
+
+#### `schema`
```

Comment:
I think it would be good to cover what can go in a `schema` by either listing them here, or linking to other documentation, or both

Reply:
You mean linking to the json schema docs?

Reply:
Yup!